### PR TITLE
Fix CodeGen command and add a unit test

### LINKED ISSUE #1643: GenerateCodeCommand bug
This command has a replacementMap dictionary, and it uses it to find the keys in the generated code template and replace them with the values. But there is one pair in the dictionary with a key that doesn't exist in the template.

### PR DESCRIPTION
**Pull request by yaeldekel**
Fixes #1643 .

### CODE REVIEW

File: src/Microsoft.ML.Api/GenerateCodeCommand.cs [hunk 1]
```diff
@@ -139,7 +139,6 @@ public void Run()
                     new Dictionary<string, string>
                     {
                         { "EXAMPLE_CLASS_DECL", loaderSb.ToString() },
-                        { "SCORED_EXAMPLE_CLASS_DECL", nonScoreSb.ToString() },
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Api/GenerateCodeCommand.cs]: >SCORED_EXAMPLE_CLASS_DECL [](start = 27, length = 25)

In our internal repo we actually have this thing

          // These are all remaining available columns, either supplied as the input, or intermediate
            // columns generated by the transforms. Materializing these columns has a performance cost,
            // so they are commented out. Feel free to uncomment any column that is useful for your scenario.
#if false
/*#SCORED_EXAMPLE_CLASS_DECL#*/
/*#/SCORED_EXAMPLE_CLASS_DECL#*/
#endif

Are you sure you want to delete it?

File: test/Microsoft.ML.TestFramework/TestCommandBase.cs [hunk 2]
```diff
@@ -2099,5 +2097,32 @@ public void Datatypes()
...
             Done();
         }
+
+        [Fact]
+        public void CommandCodeGen()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return;
+
+            // REVIEW petelu: this tests that the generated output matches the baseline. This does NOT baseline
```
Comment (Zruty0) [inline test/Microsoft.ML.TestFramework/TestCommandBase.cs]: who is petelu?..

File: src/Microsoft.ML.Api/GenerateCodeCommand.cs (hunk 1 above)
Comment (Zruty0) [reply src/Microsoft.ML.Api/GenerateCodeCommand.cs]: Yep, that's strange.

File: src/Microsoft.ML.Api/GeneratedCodeTemplate.csresource [hunk 3]
```diff
@@ -45,7 +53,7 @@ PredictAsync(modelPath);
 /// This method demonstrates how to run prediction.
 ///
 ///</summary>
-public static void Predict(string modelPath)
+public static void PredictAsync(string modelPath)
```
Comment (Zruty0) [inline src/Microsoft.ML.Api/GeneratedCodeTemplate.csresource]: >PredictAsync [](start = 19, length = 12)

Did you try running this code now? I think the return type now should be `Task`?

File: src/Microsoft.ML.Api/GeneratedCodeTemplate.csresource [hunk 4]
```diff
@@ -45,7 +53,7 @@ PredictAsync(modelPath);
...
 {
     var model = await PredictionModel.ReadAsync<InputData, ScoredOutput>(modelPath);
```
[the remark this answers is not in the document]
Comment (yaeldekel) [reply src/Microsoft.ML.Api/GeneratedCodeTemplate.csresource]: I opened issue #1738 to convert the command to use the new api.

File: test/Microsoft.ML.TestFramework/TestCommandBase.cs [hunk 5]
```diff
@@ -672,16 +672,15 @@ public void CommandShowSchema()
         }
 
         [TestCategory(Cat)]
-        [Fact(Skip = "Need CoreTLC specific baseline update")]
+        [Fact]
         public void CommandShowSchemaModel()
         {
-            string trainDataPath = GetDataPath(@"..\UCI", "adult.test.tiny");
+            string trainDataPath = GetDataPath("adult.tiny.with-schema.txt");
```
Comment (singlis) [inline test/Microsoft.ML.TestFramework/TestCommandBase.cs]: Can this be TestDataset.adult.testFilename?

File: test/Microsoft.ML.TestFramework/TestCommandBase.cs [hunk 6]
```diff
@@ -2096,5 +2093,32 @@ public void Datatypes()
...
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return;
+
+            // REVIEW: this tests that the generated output matches the baseline. This does NOT baseline
+            // the console output. Currently, there's no console output either, but if some is added, a baseline test
+            // will be in order.
+
+            // First, train a model on breast-cancer.
+            var dataPath = GetDataPath("breast-cancer.txt");
```
Comment (singlis) [inline test/Microsoft.ML.TestFramework/TestCommandBase.cs]: TestDatasets.breastCancer.testFilename?

File: test/Microsoft.ML.Predictor.Tests/TestParallelFasttreeInterface.cs [hunk 7]
```diff
@@ -173,7 +173,7 @@ public TestParallelFasttreeInterface(ITestOutputHelper helper)
         {
         }
 
-        [Fact]
+        [Fact(Skip = "'checker' is not a valid value for the 'parag' argument in FastTree")]
```
[the remark this answers is not in the document]
Comment (singlis) [reply test/Microsoft.ML.Predictor.Tests/TestParallelFasttreeInterface.cs]: If we do skip the test, should there be an issue filed on how to fix?

---
In reply to: [237247906](https://github.com/dotnet/machinelearning/pull/1654#discussion_r237247906) [](ancestors = 237247906)

**Comment by singlis** [comment]
posted a couple of comments, but overall looks good.

**Pull request merged**